Week5/Solution.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Alyssa Lingad
+ * 
+ * MICP: Week 4
+ */
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import java.util.*;
+
+
+
+public class Solution
+{
+	public static boolean isPalindrome(Node head) throws Exception{
+		if (head == null){
+			throw new Exception();
+		}
+		Node copy = copyLL(head);
+		int nodeCount = 0;
+		
+		Node prev = null;
+		Node curr = copy;
+		Node next = null;
+		while (curr != null){
+			next = curr.next;
+			curr.next = prev;
```

Comment:
You can put this entire while loop in a method that is name like "GetLinkdListLenght", and is responsible for caluclating the length of the list. (Similar to how you have a separate method for cloning the entire linked list)